convert2rhel/unit_tests/redhatrelease_test.py

```diff
@@ -85,6 +96,82 @@ def test_patch_yum_conf_existing_distroverpkg(self):
             self.assertEqual(yum_conf._yum_conf_content.count("#distroverpkg="), 1)
 
 
+@pytest.mark.parametrize(
+    "subprocess_ret, expected",
+    [
+        ("S.5....T.  c /etc/yum.conf", True),
+        ("", False),
+        (".......T.  c /etc/yum.conf", False),
+    ],
+)
+def test_is_modified_yum(monkeypatch, subprocess_ret, expected):
+
+    is_modified_run(
+        monkeypatch=monkeypatch,
+        pkg_type="yum",
+        subprocess_ret=subprocess_ret,
+        expected=expected,
+    )
+
+
+@pytest.mark.parametrize(
+    "subprocess_ret, expected",
+    [
+        ("S.5....T.  c /etc/dnf/dnf.conf", True),
+        (".......T.  c /etc/dnf/dnf.conf", False),
+        (".M.......  g /var/lib/dnf", False),
+    ],
+)
+def test_is_modified_dnf(monkeypatch, subprocess_ret, expected):
+
+    is_modified_run(
+        monkeypatch=monkeypatch,
+        pkg_type="dnf",
+        subprocess_ret=subprocess_ret,
+        expected=expected,
+    )
+
+
+def test_is_modified_unknown(monkeypatch):
+    is_modified_run(monkeypatch, "unknown", "anything", False)
+
+
+# function called by test_is_modified_yum/dnf/unknown
+def is_modified_run(monkeypatch, pkg_type, subprocess_ret, expected):
+    monkeypatch.setattr(pkgmanager, "TYPE", value=pkg_type)
+
+    run_subprocess = mock.Mock(return_value=(subprocess_ret, "anything"))
+    monkeypatch.setattr(utils, "run_subprocess", value=run_subprocess)
+
+    assert YumConf._is_modified() == expected
+
+
+@pytest.mark.parametrize("is_modified", [True, False])
+def test_patch(monkeypatch, is_modified):
```

Comment:
These tests are a bit hard to understand. The reason in that, usually test has there steps inside:
1. Setting the environment of the test function (i.e. mocking)
2. Running the testing function in this environment
3. Check expected outputs (asserting)

I understand your motivation to reuse common steps, but I think we can hit two goals at once

Reply:
try to solve the problem with one test only, something like this:
```python
@pytest.mark.parametrize(
    (
        "pkg_manager",
        "run_subprocess_returns",
        "should_raise",
        "expected_result",
    ),
    (
        (
            "yum",
            "bla bla bla",
            None,
            "bla bla bla",
        ),
        (
            "dnf",
            "bla bla bla",
            None,
            "bla bla bla",
        ),
        (
            "unknown",
            "bla bla bla",
            Exception,
            "bla bla bla",
        ),
    ),
)
def test_yum_conf_is_modified(
    pkg_manager, run_subprocess_returns, should_raise, expected_result, caplog
):
    """TODO for Adam.

    - Use caplog for asserting logs
    - should_raise is optional, and may exists here for checking that exceptions
        were raised

    """
```
And compare with your solution